core/src/main/java/org/elasticsearch/search/rescore/RescoreBuilder.java

```diff
@@ -78,23 +139,41 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
         protected abstract XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException;
 
+        @Override
+        public abstract int hashCode();
+
+        @Override
+        public abstract boolean equals(Object obj);
     }
 
     public static class QueryRescorer extends Rescorer {
+
         private static final String NAME = "query";
-        private QueryBuilder queryBuilder;
-        private Float rescoreQueryWeight;
-        private Float queryWeight;
-        private String scoreMode;
+        public static final QueryRescorer PROTOTYPE = new QueryRescorer(new MatchAllQueryBuilder());
+        public static final float DEFAULT_RESCORE_QUERYWEIGHT = 1.0f;
+        public static final float DEFAULT_QUERYWEIGHT = 1.0f;
+        public static final QueryRescoreMode DEFAULT_SCORE_MODE = QueryRescoreMode.Total;
+        private QueryBuilder<?> queryBuilder;
```

Comment:
Good point, ideally yes. But when moving the parser logic over from RescoreParseElement just now, I realized I need an additional setter for the inner query. The reason for this is that for the Rescore Element Simon replaced our well known parser logic with the more general ObjectParser, and I'd like to be able to use this and not go back to the "old style". However, ObjectParser takes an already constructed instance and then only calls setters on it, so I need to be able to re-set the queryBuilder if I want to reuse that. At least for now, I think. Maybe there's another way to make this final, will keep that in mind.